modules/apps/object/object-api/src/main/java/com/liferay/object/system/BaseSystemObjectDefinitionMetadata.java

```diff
@@ -72,6 +73,26 @@ public String getTitleObjectFieldName() {
 		return "id";
 	}
 
+	@Override
+	public long upsertBaseModel(
+			String externalReferenceCode, long companyId, User user,
+			Map<String, Object> values)
+		throws Exception {
+
+		BaseModel<?> baseModel = fetchBaseModelByExternalReferenceCode(
+			externalReferenceCode, companyId);
+
+		if (baseModel != null) {
+			long primaryKey = (Long)baseModel.getPrimaryKeyObj();
```

Comment:
Maybe better cast to long instead:

```
long primaryKey = (long)baseModel.getPrimaryKeyObj();
```

Reply:
Ups sorry, makes sense! Better to cast it using the primitive instead of the Object